src/routes/Search/Search.js

```diff
@@ -9,14 +10,21 @@ class Search extends Component {
 
   }
 
+  static contextType = UserContext
+
   state = { error: null }
 
   firstInput = React.createRef()
 
   handleSubmit = ev => {
     ev.preventDefault()
-    const { zipcode } = ev.target
-
+    const { street, city, state, zip } = ev.target
+    const address = `${street.value}, ${city.value}, ${state.value}, ${zip.value}`
+    const user = {
+      username: 'anon',
+      address,
+    }
+    this.context.user = user
```

Comment:
Is this setting a new user in the context? I think the context should have a method that takes in this new object as an argument, and in the context there's a setState({user}) function

Reply:
I think it should go as:

`
if (TokenService.hasAuthToken()) {
  this.context.setUser({
    ...this.context.user,
    address
  })
} else {
  this.context.setUser({
    username: 'anon',
    address
  })
}`